snakemake/jobs.py

```diff
@@ -1110,75 +1111,87 @@ async def postprocess(
-        )
-        wait_for_local = (
-            SharedFSUsage.STORAGE_LOCAL_COPIES
-            in self.dag.workflow.storage_settings.shared_fs_usage
-        )
-        if (
-            self.dag.workflow.exec_mode == ExecMode.SUBPROCESS
-            or shared_input_output
-            or (self.dag.workflow.remote_exec and not shared_input_output)
-            or self.is_local
-        ):
-            if not error and handle_touch:
-                self.dag.handle_touch(self)
-            if handle_log:
-                await self.dag.handle_log(self)
-            if not error:
-                await self.dag.check_and_touch_output(
-                    self,
-                    wait=self.dag.workflow.execution_settings.latency_wait,
-                    ignore_missing_output=ignore_missing_output,
-                    # storage not yet handled, just require the local files
-                    wait_for_local=True,
-                )
-            self.dag.unshadow_output(self, only_log=error)
+        begin_time = time.perf_counter()
 
+        async with asyncio.TaskGroup() as tg:
+            if self.dag.is_edit_notebook_job(self):
+                # No postprocessing necessary, we have just created the skeleton notebook and
+                # execution will anyway stop afterwards.
+                return
+
+            shared_input_output = (
+                SharedFSUsage.INPUT_OUTPUT
+                in self.dag.workflow.storage_settings.shared_fs_usage
+            )
+            wait_for_local = (
+                SharedFSUsage.STORAGE_LOCAL_COPIES
+                in self.dag.workflow.storage_settings.shared_fs_usage
+            )
             if (
-                not error
-                and self.rule.is_template_engine
-                and not is_flagged(self.output[0], "temp")
+                self.dag.workflow.exec_mode == ExecMode.SUBPROCESS
+                or shared_input_output
+                or (self.dag.workflow.remote_exec and not shared_input_output)
+                or self.is_local
             ):
-                # TODO also check if consumers are executed on the same node
-                check_template_output(self)
+                if not error and handle_touch:
+                    self.dag.handle_touch(self)
+                if handle_log:
+                    tg.create_task(self.dag.handle_log(self))
+                if not error:
+                    tg.create_task(
+                        self.dag.check_and_touch_output(
+                            self,
+                            wait=self.dag.workflow.execution_settings.latency_wait,
+                            ignore_missing_output=ignore_missing_output,
+                            # storage not yet handled, just require the local files
+                            wait_for_local=True,
+                        )
+                    )
+                self.dag.unshadow_output(self, only_log=error)
```

Comment:
I think this is problematic. The idea of the await was that the check_and_touch happens before the stuff below like the unshadow. If I don't get this wrong here, now the check and touch happens in the background, in theory in parallel to the unshadow?